src/Common/tests/System.Net/Sockets/Performance/SocketTestClient.cs

```diff
@@ -118,12 +126,20 @@ private void OnConnect()
             Receive(OnReceive);
         }
 
-        public abstract void Send(Action<int> onSendCallback);
+        public abstract void Send(Action<int, SocketError> onSendCallback);
 
         // Called when the entire _sendBuffer has been sent.
-        private void OnSend(int bytesSent)
+        private void OnSend(int bytesSent, SocketError error)
         {
-            _log.WriteLine(this.GetHashCode() + " OnSend({0})", bytesSent);
+            _log.WriteLine(this.GetHashCode() + " OnSend({0}, {1})", bytesSent, error);
+
+            // TODO: an error should fail the test.
```

Comment:
This is debatable in a perf/stress test. We could count failures and depending on resource pressure we could determine the number of acceptable failures for send/receive.
We can use native socket perf tests as a baseline.